Projects/Contracts/Sources/View/CoInsured/CoInusuredInput.swift

```diff
@@ -0,0 +1,273 @@
+import Combine
+import SwiftUI
+import hCore
+import hCoreUI
+
+struct CoInusuredInput: View, KeyboardReadable {
+    @State var fullName: String
+    @State var firstName: String
+    @State var lastName: String
+    @State var SSN: String
+    @State var type: CoInsuredInputType?
+    @State var keyboardEnabled: Bool = false
+    @State var nameFetchedFromSSN: Bool = false
+    @PresentableStore var store: ContractStore
+    let isDeletion: Bool
+    let contractId: String
+    @State var noSSN = false
+
+    public init(
+        isDeletion: Bool,
+        fullName: String?,
+        SSN: String?,
+        contractId: String
+    ) {
+        self.isDeletion = isDeletion
+        self.fullName = fullName ?? ""
+        self.firstName = ""
+        self.lastName = ""
+        self.SSN = SSN ?? ""
+        self.contractId = contractId
+    }
+
+    var body: some View {
+        hForm {
+            VStack(spacing: 4) {
+                if isDeletion {
+                    deleteCoInsuredFields
+                } else {
+                    addCoInsuredFields
+                }
+                hSection {
+                    LoadingButtonWithContent(ContractStore.self, .fetchNameFromSSN) {
+                        if !(buttonIsDisabled || nameFetchedFromSSN || noSSN) {
```

Comment:
we should move logic to one property inside newly generated VM where we will return true/false
Like in  "else if isDeletion"